### PR TITLE
add Event Npc Proxy

### PR DESCRIPTION
**Pull request by anjeshshrestha**
This adds the generic Event proxy window, can use the update to provide status similar to Labyrinth and GOP. also add api implementation

### CODE REVIEW

File: src/main/java/com/github/manolo8/darkbot/core/manager/FacadeManager.java [hunk 1]
```diff
@@ -3,21 +3,7 @@
...
-import com.github.manolo8.darkbot.core.objects.facades.EternalGateProxy;
-import com.github.manolo8.darkbot.core.objects.facades.FrozenLabyrinthProxy;
-import com.github.manolo8.darkbot.core.objects.facades.HighlightProxy;
-import com.github.manolo8.darkbot.core.objects.facades.LogMediator;
-import com.github.manolo8.darkbot.core.objects.facades.SettingsProxy;
-import com.github.manolo8.darkbot.core.objects.facades.SlotBarsProxy;
-import com.github.manolo8.darkbot.core.objects.facades.SpaceMapWindowProxy;
-import com.github.manolo8.darkbot.core.objects.facades.StatsProxy;
-import com.github.manolo8.darkbot.core.objects.facades.GauntletPlutusProxy;
+import com.github.manolo8.darkbot.core.objects.facades.*;
```
Comment (Pablete1234) [inline src/main/java/com/github/manolo8/darkbot/core/manager/FacadeManager.java:6]: disable wildcard imports in your IDE

Comment (anjeshshrestha) [reply src/main/java/com/github/manolo8/darkbot/core/manager/FacadeManager.java:6]: ahhh not again 😭

File: src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import static com.github.manolo8.darkbot.Main.API;
+
+public class NpcEventProxy extends Updatable implements NpcEventAPI {
+    public double time;
+    public String eventID;
+    public String eventName;
+    public String eventDescriptionID;
+    public String eventDescription;
+    public String npc_leftStatus;
```
Comment (Pablete1234) [inline src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java:15]:
```suggestion
    public String npcLeftStatus;
```

File: src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        return time;
+    }
+
+    @Override
+    public NpcEventAPI.Status getStatus() {
+        if(this.eventActive){
+            return Status.ACTIVE;
+        }else{
+            return Status.INACTIVE;
+        } //need to detect when event is over;
```
Comment (Pablete1234) [inline src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java:31]:
```suggestion
        if (this.eventActive) {
            return Status.ACTIVE;
        } else {
            return Status.INACTIVE;
        } //need to detect when event is over;
```
please fix spacing, also consider if we can only detect active/inactive potentially making it a boolean (ideally tho, we'd like to be able to distinguish that)

**Review by Pablete1234 — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java (hunk 3 above)
Comment (anjeshshrestha) [reply src/main/java/com/github/manolo8/darkbot/core/objects/facades/NpcEventProxy.java]: This was to keep consistency with other proxies

**Review by Pablete1234 — APPROVED** [review]

**Comment by Pablete1234** [comment]
Feel free to update the api version to 0.7.2 which should make the build pass

**Review by Pablete1234 — APPROVED** [review]

**Pull request merged**